### PR TITLE
Reset rooms array after each clients() call

### PR DESCRIPTION
**Pull request by Nibbler999**
If you call clients() multiple times with different rooms you end up with clients from all the rooms specified in previous calls.

### CODE REVIEW

File: test/socket.io.js [hunk 1]
```diff
@@ -730,6 +730,55 @@ describe('socket.io', function(){
...
+        sio.of('/other').on('connection', function(socket){
+          socket.join('foo', function() {
+            otherSid = socket.id;
+            --total || getClients();
+          });
+        });
+      });
+      function getClients() {
+        sio.of('/chat').in('foo').clients(function(error, sids) {
+          expect(error).to.be.undefined;
```
Comment (nkzawa) [inline test/socket.io.js]: I believe there is no syntax like this on `expect.js`. It should be `expect().to.be(undefined)`.

**Comment by nkzawa** [comment]
:+1: it almost looks good to me.

**Comment by Nibbler999** [comment]
Changed test to use `expect(error).to.not.be.ok()`

**Comment by rauchg** [comment]
Great find.

**Pull request closed without merging**